### PR TITLE
SQL: [Docs] Add limitation for sorting on aggs

### LINKED ISSUE #52204: SQL: [Docs] Add limitation for scalars and Agg Order By
When using aggregate functions in the `ORDER BY` then only plain
functions can be used. Operators and scalars are not allowed. Examples
of not allowed queries:
```
SELECT a, f(max(b)) GROUP BY a ORDER BY f(max(b))
SELECT a, max(b) GROUP BY a ORDER BY f(max(b))
SELECT a, max(b) + min(b) AS diff GROUP BY a ORDER BY diff
```
This limitation is missing from the docs.

### PR DESCRIPTION
**Pull request by matriv**
Add a section to point out that when ordering by an aggregate
only plain aggregate functions are allowed, no scalars/operators
can be used on top of them.

Fixes: #52204

### CODE REVIEW

File: docs/reference/sql/limitations.asciidoc [hunk 1]
```diff
@@ -118,6 +118,22 @@ SELECT * FROM test GROUP BY age ORDER BY COUNT(*) LIMIT 100;
 It is possible to run the same queries without a `LIMIT` however in that case if the maximum size (*10000*) is passed,
 an exception will be returned as {es-sql} is unable to track (and sort) all the results returned.
 
+Moreover, if there are aggregation(s) in the `ORDER BY`, they must be the exact the same exact aggregation
```
Comment (bpintea) [inline docs/reference/sql/limitations.asciidoc]: "they must be ~the exact~ the same exact aggregation"

File: docs/reference/sql/limitations.asciidoc [hunk 2]
```diff
@@ -118,6 +118,22 @@ SELECT * FROM test GROUP BY age ORDER BY COUNT(*) LIMIT 100;
...
+expressions(s), as used in the `SELECT` clause. The same applies for the fields that are defined the `GROUP BY` clause:
```
Comment (bpintea) [inline docs/reference/sql/limitations.asciidoc]: "that are defined **in** the `GROUP BY`"

File: docs/reference/sql/limitations.asciidoc [hunk 3]
```diff
@@ -118,6 +118,22 @@ SELECT * FROM test GROUP BY age ORDER BY COUNT(*) LIMIT 100;
...
+[source, sql]
+--------------------------------------------------
+SELECT age, MAX(salary) FROM test GROUP BY gender ORDER BY age % 10, MAX(salary);
+
+SELECT age, MAX(salary) FROM test GROUP BY gender ORDER BY age, CAST(MAX(salary) AS FLOAT);
+
+SELECT age % 10, MAX(salary) FROM test GROUP BY gender ORDER BY age, MAX(salary);
+
+SELECT age, MAX(salary) / 12 AS max_salary FROM test GROUP BY ORDER BY age, MAX(salary);
+--------------------------------------------------
```
Comment (bpintea) [inline docs/reference/sql/limitations.asciidoc]: Might be useful to add one-two examples on how to change the statement, so that it works. Up to you.

**Review by bpintea — APPROVED** [review]
LGTM, left some minor needed corrections.

**Comment by matriv** [comment]
Not 100% correct. Need to recheck the behaviour first.

**Comment by matriv** [comment]
Fixed the limitation to mention only the aggregate functions.
Scalars on top of the grouping keys can be applied in the ORDER BY.

**Review by astefan — APPROVED** [review]
LGTM

File: docs/reference/sql/limitations.asciidoc [hunk 4]
```diff
@@ -118,6 +118,17 @@ SELECT * FROM test GROUP BY age ORDER BY COUNT(*) LIMIT 100;
 It is possible to run the same queries without a `LIMIT` however in that case if the maximum size (*10000*) is passed,
 an exception will be returned as {es-sql} is unable to track (and sort) all the results returned.
 
+Moreover, the aggregation(s) used in the `ORDER BY`, must be only plain aggregate functions. No scalar
```
Comment (bpintea) [inline docs/reference/sql/limitations.asciidoc]: "_must be only_ plain aggregate functions" could be rephrased to "can only be ..."

**Review by bpintea — APPROVED** [review]
LGTM, only left a slight improvement optional suggestion.

**Pull request merged**